src/modules/navigator/mission.cpp

```diff
@@ -811,6 +816,11 @@ Mission::read_mission_item(bool onboard, int offset, struct mission_item_s *miss
 		return false;
 	}
 
+	if(offset==0){
```

Comment:
If we choose to use the read message differently at some point this will result in unnecessary messages. It could be in set_mission_items or advance_mission